### PR TITLE
8256594: Unexpected warning: SIGSEGV handler flags expected:SA_RESTART|SA_SIGINFO found:SA_RESTART|SA_SIGINFO

### PR DESCRIPTION
**Pull request by dholmes-ora**
In the recent POSIX signal code cleanup (JDK-8253742) I misunderstood the role of the SIGNIFICANT_SIGNAL_MASK and removing its application causes -Xcheck:jni to report errors with the signal handler due to an unexpected bit in the sigaction.sa_flags field.

This fix restores the masking of that flag, for Linux only, and more clearly indicating what it is. I also augmented the flag printing code to show when an unexpected flag is present (and so avoid the mysterious "expected X found X" message).

See bug report for more details.

Thanks.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Testing

|     | Linux aarch64 | Linux arm | Linux ppc64le | Linux x64 | Linux x86 | Windows x64 | macOS x64 |
| --- | ----- | ----- | ----- | ----- | ----- | ----- | ----- |
| Build / test | ⏳ (1/1 running) | ⏳ (1/1 running) | ⏳ (1/1 running) | ⏳ (1/1 running) | ⏳ (1/1 running) | ⏳ (1/1 running) | ⏳ (1/1 running) |

### Issue
 * [JDK-8256594](https://bugs.openjdk.java.net/browse/JDK-8256594): Unexpected warning: SIGSEGV handler flags expected:SA_RESTART|SA_SIGINFO found:SA_RESTART|SA_SIGINFO

### Reviewers
 * [Thomas Stuefe](https://openjdk.java.net/census#stuefe) (@tstuefe - **Reviewer**) ⚠️ Review applies to 50ae08472f72c9e92a578670e4411575f0529ef3
 * [Coleen Phillimore](https://openjdk.java.net/census#coleenp) (@coleenp - **Reviewer**) ⚠️ Review applies to f7626ba1fc30790af3fe7ec523b1c723d9ab6d5c
 * [Daniel D. Daugherty](https://openjdk.java.net/census#dcubed) (@dcubed-ojdk - **Reviewer**) ⚠️ Review applies to f7626ba1fc30790af3fe7ec523b1c723d9ab6d5c

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1309/head:pull/1309`
`$ git checkout pull/1309`

### CODE REVIEW

File: src/hotspot/os/posix/signals_posix.cpp
Comment (tstuefe) [inline src/hotspot/os/posix/signals_posix.cpp:668]: Maybe name this "unknown_flag_mask" ?

**Review by tstuefe — APPROVED** [review]
LGTM.

Note that a maybe more elegant approach - avoiding the double definition of known flags - would have been, in the print loop, to store the checked flags in a mask. And then test for ~mask at the final step.

The numeric comparison of flags in check_jni - which requires this SA_RESTORER_FLAG_MASK workaround - is a bit grating. Would have been better to compare just known flags. But this is old coding.

All idle nitpicking :) Patch is fine.

..Thomas

**Comment by dholmes-ora** [comment]
Can I get a second review ASAP please as this is causing quite a bit of noise in the testing.
Thanks.

**Comment by dholmes-ora** [comment]
When discussing with Coleen I realized I had omitted part of the fix -  the masking of VMError::get_resetted_sigflags(sig).

**Review by coleenp — APPROVED** [review]
So this looks like it matches the 3 places that 8253742 removed.

File: src/hotspot/os/posix/signals_posix.cpp
Comment (dcubed-ojdk) [inline src/hotspot/os/posix/signals_posix.cpp:54]: Should this comment mention `SA_RESTORER` or `SA_RESTORER_FLAG_MASK`?

**Review by dcubed-ojdk — APPROVED** [review]
Changes match what you described. Thumbs up.

File: src/hotspot/os/posix/signals_posix.cpp
Comment (dholmes-ora) [reply src/hotspot/os/posix/signals_posix.cpp:54]: I'll tweak the comment before pushing.

**Comment by dholmes-ora** [comment]
Thanks for the reviews.

/integrate

**Pull request closed without merging**
